ribbon.py

```diff
@@ -0,0 +1,545 @@
+#!/usr/bin/env python
+# coding: Latin
+
+import json
+from my_button import MyScale
+# Load all standard tools for image processing challenges
+from img_base_class import *
+import random
+import cv2.aruco as aruco
+import math
+
+# Image stream processing thread
+class StreamProcessor(threading.Thread):
+    def __init__(self, screen=None, camera=None, drive=None, dict=None):
+        super(StreamProcessor, self).__init__()
+        self.camera = camera
+        image_width, image_height = self.camera.resolution
+        self.image_centre_x = image_width / 4.0
+        self.image_centre_y = image_height / 4.0
+        self.CROP_HEIGHT = 80
+        self.drive = drive
+        self.screen = screen
+        self.stream = picamera.array.PiRGBArray(camera)
+        self.event = threading.Event()
+        self.terminated = False
+        self.small_dict = dict
+        self.MAX_AREA = 4000  # Largest target to move towards
+        self.MIN_CONTOUR_AREA = 10
+        self.MIN_MARKER_AREA = 200
+        self.ribbon_colour = 'blue'
+        self.MARKER_COLOUR = 'purple'
+        self.ALT_MARKER = 'green'
+        self.MARKERS_ON_THE_LEFT = False 
+        self.MARKER_CROP_HEIGHT = 35
+        self.MARKER_CROP_WIDTH = 100
+        self.MARKER_SPEED=0.2
+        self.found = False
+        self.retreated = False
+        self.cycle = 0
+        self.mode = [self.ribbon_following, self.marker, self.turntable, self.block_pushing]
+        self.mode_number = 0
+        self.turntable_stage= [self.approach, self.entry, self.leaving]
+        self.stage_number = 0
+        self.TURNTABLE_MARKER = 3
+        self.turntable_approached = False
+        self.APPROACH_DIST = 50
+        self.RIBBON_APPROACH_DIST = 10
+        self.APPROACH_TOL = 0.05
+        self.menu = False
+        self.last_a_error = 0
+        self.last_t_error = 0
+        self.last_before_that_t_error = 0
+        self.MAX_SPEED = 0.3
+        self.isstuck = False
```

Comment:
super minor and of absolutely no importance: snake case for variables.. `isstuck` is a bit..
![eye_twitch_krabbs](https://user-images.githubusercontent.com/1022090/39082533-a57fe060-454c-11e8-87d3-bf0a11151a26.gif)